### PR TITLE
Reflect integration state in the flow-view component

### PR DESCRIPTION
**Pull request by gashcrumb**
This makes the sidebar reflect the state of the integration as the user selects connections, the expanding items are closed and the icon available on the connection replaces the cube.

fireshot capture 1 - red hat ipaas_ - http___localhost_4200_integrations_

### CODE REVIEW

File: src/app/integrations/create-page/current-flow.service.ts [hunk 1]
```diff
@@ -69,14 +79,26 @@ export class CurrentFlow {
         log.debugc(() => 'Saving integration: ' + this._integration);
         // poor man's clone in case we need to munge the data
         const integration = JSON.parse(JSON.stringify(this._integration));
-        this.store.create(integration).toPromise().then((i: Integration) => {
+        // TODO munging connection objects for now
+        const steps = integration.steps;
+        const newSteps = [];
+        for (const step of steps) {
```
Comment (gashcrumb) [inline src/app/integrations/create-page/current-flow.service.ts]: @jimmidyson right here...

Comment (jimmidyson) [inline src/app/integrations/create-page/current-flow.service.ts]: ha nice!

**Comment by jimmidyson** [comment]
Let's [merge] this in...

**Comment by jimmidyson** [comment]
Let's [merge] this in...

**Pull request merged**